app/pages/profile.tsx

```diff
@@ -0,0 +1,228 @@
+
+import { AuthContext } from "../context/AuthContext";
+
+import NavBar from "../components/navBar";
+import UserStats from "../components/userStats";
+import { ref, uploadBytes, getDownloadURL } from "firebase/storage";
+
+import { v4 } from "uuid";
+
+function Profile() {
+  const router = useRouter();
+  const appContext = useContext(AuthContext);
+  const userData = appContext?.userData;
+  const [imageUpload, setImageUpload] = useState<File | null>();
+  const [imageError, setImageError] = useState(false);
+
+  const handleUsernameChange = async (event: any) => {
+    const newUsername = event.target.elements.newUsername.value;
+    if (appContext && appContext.currentUser) {
+      const user = appContext.currentUser;
+      await updateProfile(user, {
+        displayName: newUsername,
+      }).then(async () => {
+        await updateDoc(doc(db, "users", user.uid), {
+          "data.displayName": newUsername,
+        });
+      });
+    }
+  };
+
+  // Handler that uploads user profile image and
+  // updates user auth instance photoURL
+  const handleUploadImage = async (event: any) => {
+    if (imageUpload == null) return;
+    const imageRef = ref(storage, `profile/${imageUpload.name + v4()}`);
+    uploadBytes(imageRef, imageUpload).then(async () => {
+      // After new image is uploaded, change user display image
+      if (appContext && appContext.currentUser) {
+        const user = appContext.currentUser;
+        const newPhotoLink = await getDownloadURL(imageRef);
+        await updateProfile(user, {
+          photoURL: newPhotoLink,
+        });
+        await updateDoc(doc(db, "users", user.uid), {
+          "data.photoURL": newPhotoLink,
+        });
+      }
+    });
+    router.reload();
+  };
+
+  return (
+    <AuthRoute>
+      <NavBar></NavBar>
+      <div className="flex flex-row max-h-screen min-w-full place-content-center px-5 py-5 gap-x-20 mx-auto">
+        <div className="mt-4 flex flex-col">
+          <label htmlFor="imageUploadModal" className="modal-button">
+            <div className="relative border-2 rounded-xl">
+              <Image
+                className="rounded-xl avatar"
+                src={
+                  userData?.userPhotoLink || "https://placeimg.com/80/80/people"
+                }
+                height={200}
+                width={200}
+                alt="profile pic"
+              />
+              <div className="absolute top-0 left-0 w-full h-full bg-white rounded-xl opacity-0 flex justify-center items-center transition-opacity hover:opacity-50 duration-500">
+                <Image
+                  className=""
+                  src="/accountCircle.png"
+                  height={50}
+                  width={50}
+                  alt="uploadIcon"
+                />
+              </div>
+            </div>
+          </label>
+          <div className="mt-5 flex flex-col">
+            {userData && userData.userName ? (
```

Comment:
to make this reactive, maybe we could use `const [user, loading, error] = useAuthState(auth, options);` from firebase-react-hooks.

I need to have another look at our AuthContext tbh, we can probably just ditch the HOC and use firebase-react-hooks